solutions/src/main/java/ru/sberbank/school/task02/FxConversionServiceImpl.java

```diff
@@ -9,32 +9,43 @@
 import java.util.ArrayList;
 import java.util.List;
 
-
-
 public class FxConversionServiceImpl implements FxConversionService {
 
     ExternalQuotesService externalQuotesService;
 
     FxConversionServiceImpl(ExternalQuotesService externalQuotesService) {
+
+        if (externalQuotesService == null) {
+
+            throw new FxConversionException("ExternalQuotesService не должен быть null");
+        }
+
         this.externalQuotesService = externalQuotesService;
     }
 
     @Override
     public BigDecimal convert(ClientOperation operation, Symbol symbol, BigDecimal amount)
-            throws FxConversionException {
+            throws FxConversionException, IllegalArgumentException {
 
-        if (amount.compareTo(BigDecimal.valueOf(0)) < 0) {
-            throw new FxConversionException("Объем не может быть отрицательным");
+        if (operation == null) {
+            throw new IllegalArgumentException("Аргумент operation не может быть null");
         }
 
-        List<Quote> quotes = new ArrayList<>();
-        quotes = externalQuotesService.getQuotes(symbol);
+        if (symbol == null) {
+            throw new IllegalArgumentException("Аргумент symbol не может быть null");
+        }
 
+        if (amount == null || amount.compareTo(BigDecimal.valueOf(0.0)) <= 0) {
+            throw new IllegalArgumentException("Аргумент amount должен быть больше 0 и не null");
+        }
 
-        if (quotes.size() == 0) {
+        List<Quote> quotes = externalQuotesService.getQuotes(symbol);
+
+        if (quotes == null || quotes.size() == 0) {
             throw new FxConversionException("Список котировок ExternalQuotesService не должен быть пустой");
         }
 
+
         /**
```

Comment:
First sentence of Javadoc is missing an ending period.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a133661930>*